app/views/admin/enterprises/form/_business_details.html.haml

```diff
@@ -31,23 +31,42 @@
     = f.label :invoice_text, t('.invoice_text')
   .omega.eight.columns
     = f.text_area :invoice_text, style: "width: 100%; height: 100px;"
-
-.row
+.row{ data: { controller: 'terms-and-conditions', "terms-and-conditions-message-value": t('js.admin.enterprises.form.images.immediate_terms_and_conditions_removal_warning') } }
   .alpha.three.columns
     = f.label :terms_and_conditions, t('.terms_and_conditions')
     %i.text-big.icon-question-sign{ "data-controller": "help-modal-link", "data-action": "click->help-modal-link#open", "data-help-modal-link-target-value": "terms_and_conditions_info_modal" }
 
-  .omega.eight.columns
-    %a{ href: '{{ Enterprise.terms_and_conditions }}', target: '_blank', ng: { if: 'Enterprise.terms_and_conditions' } }
-      = '{{ Enterprise.terms_and_conditions_file_name }}'
-      = t('.uploaded_on')
-      = '{{ Enterprise.terms_and_conditions_updated_at }}'
-    .pad-top
-      = f.file_field :terms_and_conditions, accept: 'application/pdf', 'terms-and-conditions-warning' => 'true'
+  .omega.eight.columns#terms_and_conditions{data: { 'reflex-root': '#terms_and_conditions' } }
+    - if @enterprise.terms_and_conditions.attached?
+      = link_to "#{@enterprise.terms_and_conditions.blob.filename} #{ t('.uploaded_on') } #{@enterprise.terms_and_conditions.blob.created_at}", url_for(@enterprise.terms_and_conditions), target: '_blank'
+
+    %div
+      .button.small{ data: { controller: 'help-modal-link', action: 'click->help-modal-link#open', "help-modal-link-target-value": "terms_and_conditions_warning_modal" } }
+        = t('.upload')
+      %span{ data: { "terms-and-conditions-target": "filename" } }
+      = f.file_field :terms_and_conditions, accept: 'application/pdf', style: 'display: none;', data: { "terms-and-conditions-target": "fileinput" }
+      
     .pad-top
-      %a.button.red{ href: '', ng: {click: 'removeTermsAndConditions()', if: 'Enterprise.terms_and_conditions'} }
+    - if @enterprise.terms_and_conditions.attached?
+      %a.button.red{ href: '#', data: {  action: 'click->terms-and-conditions#remove', "terms-and-conditions-message-value": t('js.admin.enterprises.form.images.immediate_terms_and_conditions_removal_warning'), 'enterprise-id': @enterprise.id}}
```

Comment:
It seems to me that the "delete" button should be closer to the name of the file that it deletes.

Looking at [this screenshot](https://user-images.githubusercontent.com/49817236/201351670-ace8c9ed-87d8-47be-b952-d677de0bd33c.png) I found, I think it should be on the same line as the "upload" button, so it doesn't take up another line.

I think it's as simple as removing line 49 (`.pad-top`) and indenting lines 50-51. GitHub won't let me use the suggestion feature here.. oh well we can leave it as it is.

Reply:
I have another proposal: 

<img width="685" alt="Capture d’écran 2023-03-08 à 10 18 34" src="https://user-images.githubusercontent.com/296452/223673729-de9feaa6-0b4f-4553-86e6-ec01f2f14a86.png">

I've made the remove action a link (not a button) and moved it closer to the filename (because it actually remove this one, not the just uploaded one)

What do you think?

Reply:
So much better!